### PR TITLE
Library logic filter for developer

### PR DESCRIPTION
**Pull request by Serge45**
## Brief ##
Added ~~`--dev-logic-filter`~~ `--logic-yaml-filter` to `install.sh` for developer. It's equivalent to the following glob pattern for logic searching:

```bash
<logic-root>/**/<logic-yaml-filter>.yaml
```
E.g.
```bash
bash ./install.sh -a gfx90a --logic-yaml-filter *BBS*
```
then only the following logics will be built

```bash   library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/110CU/GridBased/aldebaran_Cijk_Ailk_Bljk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/110CU/GridBased/aldebaran_Cijk_Ailk_Bjlk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/110CU/GridBased/aldebaran_Cijk_Alik_Bjlk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/110CU/GridBased/aldebaran_Cijk_Alik_Bljk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/Equality/aldebaran_Cijk_Alik_Bljk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Ailk_Bljk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Ailk_Bjlk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Alik_Bjlk_BBS_BH_Bias_AH_SAV.yaml
library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aldebaran/104CU/GridBased/aldebaran_Cijk_Alik_Bljk_BBS_BH_Bias_AH_SAV.yaml
```

## Implementation ##
 - [x] Added ~~`--dev-logic-filter`~~ `--logic-yaml-filter` option to `install.sh`
 - [x] Added `--logic-filter` option to `TensileCreateLibrary.py`
 - [x] CMake integration

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -517,6 +518,9 @@ while true; do
         --legacy_hipblas_direct)
             legacy_hipblas_direct=true
             shift ;;
+        --dev_logic_filter|--dev-logic-filter)
+            logic_filter=${2}
+            shift 2;;
         --) shift ; break ;;
```
Comment (solaslin) [inline install.sh]: Oh, I think we can update the option description in --help messages

**Review by solaslin — COMMENTED** [review]
LGTM. And I'm thinking if we put #1074 together in this or separately.

**Comment by solaslin** [comment]
@Serge45 
By the way, if I use
`./install.sh -a gfx942 --dev-logic-filter *gfx942/Equality*BBS* `
Does this also achieve path filter ? Which means it extracts all logics with path containing the pattern **`gfx942/Equality/...BBS...`**

If so, then I think #1074 is a duplicated one and we can close that, use this PR directly.

**Comment by Serge45** [comment]
> @Serge45 By the way, if I use `./install.sh -a gfx942 --dev-logic-filter *gfx942/Equality*BBS* ` Does this also achieve path filter ? Which means it extracts all logics with path containing the pattern **`gfx942/Equality/...BBS...`**
> 
> If so, then I think #1074 is a duplicated one and we can close that, use this PR directly.

~~For the pattern you mentioned above, seems current implementation is not able to properly interpret the final paths. Need more investigations.~~
Update: You should use this pattern `*gfx942/Equality/*BBS*`.
Here's the result:
```bash
# LogicFilter: /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/**/*gfx942/Equality/*BBS*.yaml
# LibraryLogicFiles:
#   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bjlk_BBS_BH_Bias_AS_SAV_UserArgs.yaml
#   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Alik_Bljk_BBS_BH_UserArgs.yaml
#   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bljk_BBS_BH_Bias_AS_SAV_UserArgs.yaml
#   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bljk_BBS_BH_UserArgs.yaml
#   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bjlk_BBS_BH_UserArgs.yaml
#   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Alik_Bljk_BBS_BH_Bias_AS_SAV_UserArgs.yaml
```

File: install.sh (hunk 1 above)
Comment (Serge45) [reply install.sh]: Will amend this.

Comment (jichangjichang) [reply install.sh]: just a recommendation: revise the naming as "logic_yaml_filter"

**Comment by solaslin** [comment]
> > @Serge45 By the way, if I use `./install.sh -a gfx942 --dev-logic-filter *gfx942/Equality*BBS* ` Does this also achieve path filter ? Which means it extracts all logics with path containing the pattern **`gfx942/Equality/...BBS...`**
> > If so, then I think #1074 is a duplicated one and we can close that, use this PR directly.
> 
> ~For the pattern you mentioned above, seems current implementation is not able to properly interpret the final paths. Need more investigations.~ Update: You should use this pattern `*gfx942/Equality/*BBS*`. Here's the result:
> 
> ```shell
> # LogicFilter: /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/**/*gfx942/Equality/*BBS*.yaml
> # LibraryLogicFiles:
> #   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bjlk_BBS_BH_Bias_AS_SAV_UserArgs.yaml
> #   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Alik_Bljk_BBS_BH_UserArgs.yaml
> #   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bljk_BBS_BH_Bias_AS_SAV_UserArgs.yaml
> #   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bljk_BBS_BH_UserArgs.yaml
> #   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Ailk_Bjlk_BBS_BH_UserArgs.yaml
> #   /data0/sergelu/hipBLASLt/library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/aquavanjaram/gfx942/Equality/aquavanjaram_Cijk_Alik_Bljk_BBS_BH_Bias_AS_SAV_UserArgs.yaml
> ```

Sounds greet ! Then I will close #1074 since this PR can do both !

**Comment by imcarsonliao** [comment]
could you also add this description to documentation and readme.

File: tensilelite/Tensile/TensileCreateLibrary.py [hunk 2]
```diff
@@ -1304,6 +1305,9 @@ def splitExtraParameters(par):
   argParser.add_argument("--keep-build-tmp", dest="KeepBuildTmp", action="store_true",
                           default=False, help="Do not remove the temporary build directory (may required hundreds of GBs of space)"),
   argParser.add_argument("--validate-library", dest="ValidateLibrary", action="store_true", default=False)
+  argParser.add_argument("--logic-filter", dest="LogicFilter", action="store", default="*", type=str,
+                        help="Cutomsized logic filter, default is *, i.e. all logics."
+                        " Example: gfx942/GridBased/* for building equality of gfx942 only")
```
Comment (jichangjichang) [inline tensilelite/Tensile/TensileCreateLibrary.py]: correct the description

**Review by jichangjichang — APPROVED** [review]
please use squash merge

File: tensilelite/Tensile/TensileCreateLibrary.py (hunk 2 above)
Comment (Serge45) [reply tensilelite/Tensile/TensileCreateLibrary.py]: Fixed.

**Review by solaslin — APPROVED** [review]
Looks Great To Me

**Pull request merged**